src/App.tsx

```diff
@@ -1,18 +1,38 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42)
-  .map(n => `Item ${n}`);
+const totalItems = getNumbers(1, 42).map((n) => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [itemsPerPage, setItemsPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const numberOfItems = totalItems.length;
+
+  const indexOfFirstItemOnPage = (currentPage - 1) * itemsPerPage;
+  const indexOfLastItemOnPage = indexOfFirstItemOnPage + itemsPerPage;
+
+  const selectedItems = totalItems.slice(
+    indexOfFirstItemOnPage,
```

Comment:
Consider using a constant variable for the initial state of itemsPerPage (5) to improve code readability and maintainability.